# Entregable 1 - Checkpoint y MD5

Notebook consolidado de entrega final para Situacion 2. Raiz evaluable unica: `/workspace/geovision-cali-hf/Entrega_Final/Situacion2`.

## Objetivo

Consolidar el checkpoint `.pt` entrenado, sus metricas principales y la verificacion MD5 reproducible. Las rutas usadas aqui son las rutas finales autoritativas de la entrega.

In [1]:
from pathlib import Path
import json
import pandas as pd
import matplotlib.pyplot as plt
from IPython.display import display, Image, Markdown

ROOT = Path('/workspace/geovision-cali-hf/Entrega_Final/Situacion2')
pd.set_option('display.max_colwidth', 160)


In [2]:
with open(ROOT / 'metricas/sweep_best_metrics.json') as f:
    sweep = json.load(f)
with open(ROOT / 'metricas/sweep_best_md5.json') as f:
    md5 = json.load(f)
with open(ROOT / 'metricas/full_logging_manifest.json') as f:
    full = json.load(f)
with open(ROOT / 'metricas/full_logging_md5.json') as f:
    full_md5 = json.load(f)

summary = pd.DataFrame([
    {'campo': 'checkpoint', 'valor': 'checkpoints/sweep_best.pt'},
    {'campo': 'md5_calculado', 'valor': md5['calculated_md5']},
    {'campo': 'md5_esperado', 'valor': md5['expected_md5']},
    {'campo': 'md5_match', 'valor': md5['md5_match']},
    {'campo': 'Recall@1 test', 'valor': sweep['test']['recall_at_1_image_to_text']},
    {'campo': 'Macro Recall test', 'valor': sweep['test']['macro_recall']},
    {'campo': 'Sparsity SAE test', 'valor': sweep['test']['sae_visual_sparsity_ratio']},
    {'campo': 'MSE reconstruccion val', 'valor': sweep['val']['sae_visual_recon_mse']},
    {'campo': 'MSE reconstruccion test adicional', 'valor': sweep['test']['sae_visual_recon_mse']},
])
display(summary)


,campo,valor
0,checkpoint,checkpoints/sweep_best.pt
1,md5_calculado,2a512b70e4a837a0c652aa71f2c970b4
2,md5_esperado,2a512b70e4a837a0c652aa71f2c970b4
3,md5_match,True
4,Recall@1 test,0.604255
5,Macro Recall test,0.610207
6,Sparsity SAE test,0.880859
7,MSE reconstruccion val,0.010615
8,MSE reconstruccion test adicional,0.011387


In [3]:
comparison = pd.DataFrame([
    {
        'checkpoint': 'sweep_best.pt',
        'rol_en_entrega': 'resultado principal',
        'recall_at_1_test': sweep['test']['recall_at_1_image_to_text'],
        'macro_recall_test': sweep['test']['macro_recall'],
        'md5': md5['calculated_md5'],
        'md5_match': md5['md5_match'],
    },
    {
        'checkpoint': 'full_logging_best.pt',
        'rol_en_entrega': 'evidencia instrumentada con curvas',
        'recall_at_1_test': full['final_metrics']['test']['recall_at_1_image_to_text'],
        'macro_recall_test': full['final_metrics']['test']['macro_recall'],
        'md5': full_md5['checkpoint_best_md5'],
        'md5_match': full_md5['md5_match'],
    },
])
display(comparison)


,checkpoint,rol_en_entrega,recall_at_1_test,macro_recall_test,md5,md5_match
0,sweep_best.pt,resultado principal,0.604255,0.610207,2a512b70e4a837a0c652aa71f2c970b4,True
1,full_logging_best.pt,evidencia instrumentada con curvas,0.548936,0.550672,461bd40ef53051c9b6fca15685fee3d5,True


## Interpretacion

El checkpoint `sweep_best.pt` es el resultado principal porque fue seleccionado por el sweep de semillas e hiperparametros y alcanza `Recall@1 test = 0.6043`, por encima del minimo de rubrica `0.45`. El checkpoint `full_logging_best.pt` no reemplaza al principal; se conserva porque registra historia completa de entrenamiento y permite defender las curvas solicitadas. Ambos MD5 coinciden, por lo que los archivos `.pt` son verificables y reproducibles. Se omite la tabla detallada por clase para no mezclar KPIs globales finales con metricas diagnosticas de clase.